Cubical/Categories/Category.agda

```diff
@@ -24,59 +24,97 @@ module Cubical.Categories.Category where
+    ⋆IdR : ∀ {x y} (f : Hom[ x , y ]) → f ⋆ (id y) ≡ f
+    ⋆Assoc : ∀ {u v w x} (f : Hom[ u , v ]) (g : Hom[ v , w ]) (h : Hom[ w , x ]) → (f ⋆ g) ⋆ h ≡ f ⋆ (g ⋆ h)
+
+  -- composition: alternative to diagramatic order
+  _∘_ : ∀ {x y z} (g : Hom[ y , z ]) (f : Hom[ x , y ]) → Hom[ x , z ]
+  g ∘ f = f ⋆ g
+
+open Precategory
+
+
+-- Helpful syntax/notation
+
+_[_,_] : (C : Precategory ℓ ℓ') → (x y : C .ob) → Type ℓ'
+_[_,_] = Hom[_,_]
+
+-- needed to define this in order to be able to make the subsequence syntax declaration
+seq' : ∀ (C : Precategory ℓ ℓ') {x y z} (f : C [ x , y ]) (g : C [ y , z ]) → C [ x , z ]
+seq' = _⋆_
+
+infix 15 seq'
+syntax seq' C f g = f ⋆⟨ C ⟩ g
+
+-- composition
+comp' : ∀ (C : Precategory ℓ ℓ') {x y z} (g : C [ y , z ]) (f : C [ x , y ]) → C [ x , z ]
+comp' = _∘_
+
+infix 16 comp'
+syntax comp' C g f = g ∘⟨ C ⟩ f
+
 
-open Precategory public
 
 -- Categories
 
-record isCategory {ℓ ℓ'} (𝒞 : Precategory ℓ ℓ') : Type (ℓ-max ℓ ℓ') where
+record isCategory (C : Precategory ℓ ℓ') : Type (ℓ-max ℓ ℓ') where
   field
-    homIsSet : ∀ {x y} → isSet (𝒞 .hom x y)
+    isSetHom : ∀ {x y} → isSet (C [ x , y ])
 
-open isCategory public
 
 -- Isomorphisms and paths in precategories
 
-record CatIso {ℓ ℓ' : Level} {𝒞 : Precategory ℓ ℓ'} (x y : 𝒞 .ob) : Type ℓ' where
+record CatIso {C : Precategory ℓ ℓ'} (x y : C .Precategory.ob) : Type ℓ' where
   constructor catiso
   field
-    h : 𝒞 .hom x y
-    h⁻¹ : 𝒞 .hom y x
-    sec : 𝒞 .seq h⁻¹ h ≡ 𝒞 .idn y
-    ret : 𝒞 .seq h h⁻¹ ≡ 𝒞 .idn x
+    mor : C [ x , y ]
+    inv : C [ y , x ]
+    sec : inv ⋆⟨ C ⟩ mor ≡ C .id y
+    ret : mor ⋆⟨ C ⟩ inv ≡ C .id x
 
-pathToIso : {ℓ ℓ' : Level} {𝒞 : Precategory ℓ ℓ'} (x y : 𝒞 .ob) (p : x ≡ y) → CatIso {𝒞 = 𝒞} x y
-pathToIso {𝒞 = 𝒞} x y p = J (λ z _ → CatIso x z) (catiso (𝒞 .idn x) idx (𝒞 .seq-λ idx) (𝒞 .seq-λ idx)) p
+pathToIso : {C : Precategory ℓ ℓ'} (x y : C .ob) (p : x ≡ y) → CatIso {C = C} x y
+pathToIso {C = C} x y p = J (λ z _ → CatIso x z) (catiso (C .id x) idx (C .⋆IdL idx) (C .⋆IdL idx)) p
   where
-    idx = 𝒞 .idn x
+    idx = C .id x
 
 -- Univalent Categories
 
-record isUnivalent {ℓ ℓ'} (𝒞 : Precategory ℓ ℓ') : Type (ℓ-max ℓ ℓ') where
+record isUnivalent (C : Precategory ℓ ℓ') : Type (ℓ-max ℓ ℓ') where
   field
-    univ : (x y : 𝒞 .ob) → isEquiv (pathToIso {𝒞 = 𝒞} x y)
+    univ : (x y : C .ob) → isEquiv (pathToIso {C = C} x y)
 
 open isUnivalent public
 
 -- Opposite Categories
 
-_^op : ∀ {ℓ ℓ'} → Precategory ℓ ℓ' → Precategory ℓ ℓ'
-(𝒞 ^op) .ob = 𝒞 .ob
-(𝒞 ^op) .hom x y = 𝒞 .hom y x
-(𝒞 ^op) .idn = 𝒞 .idn
-(𝒞 ^op) .seq f g = 𝒞 .seq g f
-(𝒞 ^op) .seq-λ = 𝒞 .seq-ρ
-(𝒞 ^op) .seq-ρ = 𝒞 .seq-λ
-(𝒞 ^op) .seq-α f g h = sym (𝒞 .seq-α _ _ _)
+_^op : Precategory ℓ ℓ' → Precategory ℓ ℓ'
+(C ^op) .ob = C .ob
+(C ^op) .Hom[_,_] x y = C .Hom[_,_] y x
+(C ^op) .id = C .id
+(C ^op) ._⋆_ f g = C ._⋆_ g f
+(C ^op) .⋆IdL = C .⋆IdR
+(C ^op) .⋆IdR = C .⋆IdL
+(C ^op) .⋆Assoc f g h = sym (C .⋆Assoc _ _ _)
+
```

Comment:
Can you add a sanity check that C ^op ^op = C? Should be something like

```
involutiveOp : (C : Precategory ℓ ℓ') → C ^op ^op ≡ C
involutiveOp C = refl
```

(maybe you need to pattern match)

Reply:
Yep, I'd put it in a future PR, I'll add it here as well. Only thing is it I had to enable eta equality for it to go through directly judgmentally.

Reply:
Ah right, can you add a comment about it needing eta?

Reply:
Yep, it should be there in the latest commit